taskrunner/Services/LotteryCleanupService.cs

```diff
@@ -0,0 +1,127 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using SCJ.Booking.Data;
+using SCJ.Booking.TaskRunner.Utils;
+using Serilog;
+using Task = System.Threading.Tasks.Task;
+
+namespace SCJ.Booking.TaskRunner.Services
+{
+    public class LotteryCleanupService
+    {
+        private readonly ApplicationDbContext _dbContext;
+        private readonly ILogger _logger;
+        private readonly IConfiguration _configuration;
+
+        public LotteryCleanupService(IConfiguration configuration, ApplicationDbContext dbContext)
+        {
+            _dbContext = dbContext;
+            _logger = LogHelper.GetLogger(configuration);
+            _configuration = configuration;
+        }
+
+        /// <summary>
+        ///    Removes names and phone numbers from processed lottery entries
+        /// </summary>
+        public async Task RemovePersonalInfo()
+        {
+            const string anonymizedName = "_anonymized";
+            const string anonymizedEmail = "_anonymized@example.com";
+
+            _logger.Information("Removing personal info from lottery entries");
+
+            // calculate the oldest lottery entry to keep
+            int maxDays = _configuration.GetValue<int>(
+                "AppSettings:PurgeLotteryContactInfoAfterDays"
+            );
+            DateTime oldestDate = DateTime.Now.AddDays(-maxDays);
+
+            // get entries from ScTrialBookingRequests where ProcessingTimestamp is older than oldestDate
+            var entriesToUpdate = await _dbContext
+                .ScTrialBookingRequests.Where(e => e.ProcessingTimestamp < oldestDate)
+                .Where(e => e.Email != anonymizedEmail && e.RequestedByName != anonymizedName)
+                .ToListAsync();
+
+            if (entriesToUpdate.Count == 0)
+            {
+                _logger.Information("No entries to update");
+                return;
+            }
+
+            _logger.Information(
+                "Updating {count} old entries from ScTrialBookingRequests with anonymized details",
+                entriesToUpdate.Count
+            );
+
+            // Update each entry
+            foreach (var entry in entriesToUpdate)
+            {
+                entry.Email = anonymizedEmail;
+                entry.RequestedByName = anonymizedName;
+                entry.Phone = "";
+            }
+
+            // Save changes to the database
+            _dbContext.SaveChanges();
```

Comment:
Visual Studio is telling me there are 3 calls to `_dbContext.SaveChanges();` that should all be using `await _dbContext.SaveChangesAsync();` instead.

Reply:
Updated ✅